plugins/shadow-block-converter/src/shadow_block_converter.ts

```diff
@@ -164,15 +184,15 @@ export function shadowBlockConversionChangeListener(event) {
     const shadowBlock = shadowBlocks[i];
 
     // If connected blocks need to be converted too, add them to the list.
-    if (shadowBlock.outputConnection != null &&
-        shadowBlock.outputConnection.isConnected() &&
-        shadowBlock.outputConnection.targetBlock().isShadow()) {
-      shadowBlocks.push(shadowBlock.outputConnection.targetBlock());
+    const outputBlock: Block | null =
+        shadowBlock.outputConnection?.targetBlock();
+    const previousBlock: Block | null =
+        shadowBlock.previousConnection?.targetBlock();
+    if (outputBlock != null && outputBlock.isShadow()) {
```

Comment:
Can this be `if (outputBlock?.isShadow())`?

Reply:
Done.